SVDBasisGenerator.h

```diff
@@ -107,13 +107,9 @@ class SVDBasisGenerator
          if (getNumBasisTimeIntervals() > 0 &&
              d_svdsampler->isNewTimeInterval()) {
             d_svdsampler->resetDt(dt);
-            // YC: commenting this out unless someone think this is necessary
-            //     we call writeBasis in "endSamples()" function below.
```

Comment:
It seems that the functionality was here, we just needed to uncomment this.

Reply:
I believe that BasisWriter.C puts everything into one hdf5 file, but puts each interval in its own dataset, like the following: sprintf(tmp, "spatial_basis_%06d", d_num_intervals_written); where the variable represents the current time interval.

Reply:
I looked into basisWriter->writeBasis. It calls "getSpatialBasis" within, which runs the computeSVD. So we would only need to take samples, with this automatically writing out the basis for every interval, except the last, which "endSamples" will cover since it also calls writeBasis.